salt/modules/aptpkg.py

```diff
@@ -1659,8 +1659,15 @@ def list_repo_pkgs(*args, **kwargs):  # pylint: disable=unused-import
         salt '*' pkg.list_repo_pkgs
         salt '*' pkg.list_repo_pkgs foo bar baz
     '''
+    if args:
+        # get only information about packages in args
+        cmd = ['apt-cache', 'show' ] + [arg for arg in args]
+    else:
+        # get information about all available packages
+        cmd = ['apt-cache', 'show', '.']
```

Comment:
```python
cmd = ['apt-cache', 'show' ] + [arg for arg in args or ['.']]
```

But I don't like you use `apt-cache show .` here. It is super-slow, apparently, and doesn't speeds this up. If you use `apt-cache dump`, it will get all packages ~100Mb data about 2-3 seconds. The other one will do really long. The only reason I see to use `apt-cache show` is when you want to see specific packages, so no "skip thing" logic you've deleted before. I would then use both commands: "dump" for just everything and "show" only for specific packages:

```python
cmd = ['apt-cache']
if args:
    cmd.expand(['show'] + [arg for arg in args])
else:
    cmd.append('dump')
```

Reply:
Here is the proof: I don't know how this is ever good, when the entire core of the CPU is 100% full for another couple of minutes.

```
$ time apt-cache show . > /dev/null

real    2m24.267s
user    0m32.576s
sys     1m49.792s

```
And other story with just dumping it:
```
$ time apt-cache dump > /dev/null                                                           

real    0m1.182s
user    0m0.852s
sys     0m0.324s
```

This is just no-go.